File: dados.cpp
#include <iostream>
#include <cstdlib>  // Para srand y rand
#include <ctime>    // Para time

using namespace std;

int main() {
    srand((time(0)));  // Inicializa la semilla para la generación de números aleatorios

    char tiroini;
    const int max = 6;  // Valor máximo en el dado
    int dado, dado2, totaldado, mismo;
    bool conti = false;

    cout << "Bienvenido al CASINO GANAMUCHO" << endl;
    cout << "Desea lanzar su tiro inicial SI (a); NO (b)" << endl;
    cin >> tiroini;

    if (tiroini == 'a' || tiroini == 'A') {
        dado = rand() % max + 1; 
        dado2 = rand() % max + 1; 
        totaldado = dado + dado2;
        cout << "Sacaste: " << totaldado << endl;
        mismo = totaldado;
        
        if (totaldado == 2 || totaldado == 3 || totaldado == 12) {
            cout << "Usted ha perdido" << endl;
            cout << "Reinicie para reintentar el juego" << endl;
        } else if (totaldado == 7 || totaldado == 11) {
            cout << "Usted ha ganado" << endl;
            cout << "Felicitaciones" << endl;
        } else {

            dado = rand() % max + 1; 
            dado2 = rand() % max + 1; 
            totaldado = dado + dado2;
            cout << "Sacaste " << totaldado << endl;
            if (totaldado == mismo) {
                cout << "Usted ha ganado porque ha sacado el mismo numero del primer lanzamiento " << mismo << endl;
            } else if (totaldado == 7) {
                cout << "Usted ha perdido ha sacado " << totaldado << endl;
            } else {

                dado = rand() % max + 1;
                dado2 = rand() % max + 1;
                totaldado = dado + dado2;
                cout << "Sacaste " << totaldado << endl;
                if (totaldado == mismo) {
                    cout << "Usted ha ganado porque ha sacado el mismo numero del primer lanzamiento " << mismo << endl;
                } else if (totaldado == 7) {
                    cout << "Usted ha perdido ha sacado " << totaldado << endl;
                } else {

                    dado = rand() % max + 1; 
                    dado2 = rand() % max + 1; 
                    totaldado = dado + dado2;
                    cout << "Sacaste " << totaldado << endl;
                    if (totaldado == mismo) {
                        cout << "Usted ha ganado porque ha sacado el mismo numero del primer lanzamiento " << mismo << endl;
                    } else if (totaldado == 7) {
                        cout << "Usted ha perdido ha sacado " << totaldado << endl;
                    } else {

                        dado = rand() % max + 1; 
                        dado2 = rand() % max + 1; 
                        totaldado = dado + dado2;
                        cout << "Sacaste " << totaldado << endl;
                        if (totaldado == mismo) {
                            cout << "Usted ha ganado porque ha sacado el mismo numero del primer lanzamiento " << mismo << endl;
                        } else if (totaldado == 7) {
                            cout << "Usted ha perdido ha sacado " << totaldado << endl;
                        } 
                    }
                }
            }
        }
    } else {
        cout << "Hasta la Proxima" << endl;
    }

    return 0;
}

File: dados.py
import random
import time

random.seed(time.time())  # Inicializa la semilla para la generación de números aleatorios

tiroini = input("Bienvenido al CASINO GANAMUCHO\nDesea lanzar su tiro inicial SI (a); NO (b): ")

if tiroini.lower() == 'a':
    max_val = 6 
    dado = random.randint(1, max_val)
    dado2 = random.randint(1, max_val)
    totaldado = dado + dado2
    print(f"Sacaste: {totaldado}")
    mismo = totaldado
    

    if totaldado in {2, 3, 12}:
        print("Usted ha perdido")
        print("Reinicie para reintentar el juego")
    elif totaldado in {7, 11}:
        print("Usted ha ganado")
        print("Felicitaciones")
    else:

        dado = random.randint(1, max_val)
        dado2 = random.randint(1, max_val)
        totaldado = dado + dado2
        print(f"Sacaste {totaldado}")

        if totaldado == mismo:
            print(f"Usted ha ganado porque ha sacado el mismo número del primer lanzamiento {mismo}")
        elif totaldado == 7:
            print(f"Usted ha perdido ha sacado {totaldado}")
        else:
   
            dado = random.randint(1, max_val)
            dado2 = random.randint(1, max_val)
            totaldado = dado + dado2
            print(f"Sacaste {totaldado}")

            if totaldado == mismo:
                print(f"Usted ha ganado porque ha sacado el mismo número del primer lanzamiento {mismo}")
            elif totaldado == 7:
                print(f"Usted ha perdido ha sacado {totaldado}")
            else:
    
                dado = random.randint(1, max_val)
                dado2 = random.randint(1, max_val)
                totaldado = dado + dado2
                print(f"Sacaste {totaldado}")

                if totaldado == mismo:
                    print(f"Usted ha ganado porque ha sacado el mismo número del primer lanzamiento {mismo}")
                elif totaldado == 7:
                    print(f"Usted ha perdido ha sacado {totaldado}")
                else:

                    dado = random.randint(1, max_val)
                    dado2 = random.randint(1, max_val)
                    totaldado = dado + dado2
                    print(f"Sacaste {totaldado}")

                    if totaldado == mismo:
                        print(f"Usted ha ganado porque ha sacado el mismo número del primer lanzamiento {mismo}")
                    elif totaldado == 7:
                        print(f"Usted ha perdido ha sacado {totaldado}")
else:
    print("Hasta la Proxima")
